src/lib/validateAndCoerceTypes.ts

```diff
@@ -226,6 +231,92 @@ function disallowAdditionalProps(show = false) {
 
 if (process.env.NODE_ENV === "test") disallowAdditionalProps();
 
+const handleResultError = (
+  e: TransformDecodeError | TransformDecodeCheckError,
+  options: ValidationOptions
+) => {
+  const title = e.schema.title;
+  if (options.logErrors) {
+    console.log(JSON.stringify(e, null, 2));
+    console.log(`
+    This may happen intermittently and you should catch errors appropriately.
+    However:  1) if this recently started happening on every request for a symbol
+    that used to work, Yahoo may have changed their API.  2) If this happens on
+    every request for a symbol you've never used before, but not for other
+    symbols, you've found an edge-case (OR, we may just be protecting you from
+    "bad" data sometimes stored for e.g. misspelt symbols on Yahoo's side).
+    Please see if anyone has reported this previously:
+    
+      ${pkg.repository}/issues?q=is%3Aissue+${title}
+    
+    or open a new issue (and mention the symbol):  ${pkg.name} v${pkg.version}
+    
+      ${pkg.repository}/issues/new?labels=bug%2C+validation&template=validation.md&title=${title}
+    
+    For information on how to turn off the above logging or skip these errors,
+    see https://github.com/gadicc/node-yahoo-finance2/tree/devel/docs/validation.md.
+    
+    At the end of the doc, there's also a section on how to
+    [Help Fix Validation Errors](https://github.com/gadicc/node-yahoo-finance2/blob/devel/docs/validation.md#help-fix)
+    in case you'd like to contribute to the project.  Most of the time, these
+    fixes are very quick and easy; it's just hard for our small core team to keep up,
+    so help is always appreciated!
+    `);
+  }
+  throw e;
+};
+
+const handleOptionsError = (
+  e: TransformDecodeCheckError | TransformDecodeError,
+  { logOptionsErrors }: ValidationOptions
+) => {
+  if (logOptionsErrors) {
+    console.error(
+      `[yahooFinance] Invalid options ("${JSON.stringify(e.error, null, 2)}")`
+    );
+  }
+  throw new InvalidOptionsError("Validation called with invalid options");
+};
+
+export const validateAndCoerceTypebox = <T>({
+  type,
+  data,
+  schema,
+  options,
+}: {
+  type?: "result" | "options";
+  data: unknown;
+  schema: TSchema;
+  options: ValidationOptions;
+}): T => {
+  try {
+    return Value.Decode(schema, data);
+  } catch (e) {
+    if (
+      e instanceof TransformDecodeError ||
+      e instanceof TransformDecodeCheckError
+    ) {
+      switch (type) {
+        case "result": {
+          handleResultError(e, options);
+          break;
+        }
+        // TODO: The existing implementation of 'validate' assumes that this parameter may not be provided
```

Comment:
This is another weird thing I found. The implementation in `validate` uses a bare `else` to cover the condition that `type !== 'result'` instead of checking for `type === 'options'`. I'm guessing this is a catch all case for when someone ignores the TS and passes through completely garbage options. 

I have two thoughts on this:
1. I *feel* like this case shouldn't really exist as this function *should* only be called internally within the library where we can assume TS.
2. If we do want to be defensive around this I think it would make more sense to hoist the handling of this out of the `validateAndCoerce` function. Ideally we should be able to do that in the modules themselves and then we can have a clean interface for internal consumption, and catch weird errors where they originate rather than deep in the library.

Just my $0.02. I concede there is probably something going on here that I don't quite grok.